build.sbt

```diff
@@ -33,26 +33,82 @@ inThisBuild(
   )
 )
 
-scalacOptions ++= Seq(
-  "-deprecation",
-  "-Wunused:all",
-  "-feature",
-  "-unchecked",
-  "-Xcheck-macros",
-  "-Xprint-suspension",
-  "-Xsemanticdb",
-  "-Yexplicit-nulls",
-  "-Ysafe-init",
-  "-source:future",
-  "-Ykind-projector",
-  "-Vprofile"
-)
+lazy val slinc = project
+  .in(file("slinc"))
+  .settings(
+    name := "slinc",
+    scalacOptions ++= Seq(
+      "-deprecation",
+      "-Wunused:all",
+      "-feature",
+      "-unchecked",
+      "-Xcheck-macros",
+      "-Xprint-suspension",
+      "-Xsemanticdb",
+      "-Yexplicit-nulls",
+      "-Ysafe-init",
+      "-source:future",
+      "-Ykind-projector",
+      "-Vprofile"
+    ),
+    libraryDependencies += "org.scalameta" %% "munit" % "1.0.0-M10" % Test,
+    libraryDependencies += "org.scalameta" %% "munit-scalacheck" % "1.0.0-M10" % Test,
+    libraryDependencies += "org.scala-lang" %% "scala3-staging" % scalaVersion.value,
+    Compile / doc / scalacOptions ++= Seq("-siteroot", "docs"),
+    mimaPreviousArtifacts := previousStableVersion.value
+      .map(organization.value %% moduleName.value % _)
+      .toSet
+  )
 
-libraryDependencies += "org.scalameta" %% "munit" % "1.0.0-M10" % Test
-libraryDependencies += "org.scalameta" %% "munit-scalacheck" % "1.0.0-M10" % Test
-libraryDependencies += "org.scala-lang" %% "scala3-staging" % scalaVersion.value
+lazy val copyNative = TaskKey[Set[File]]("copy shared lib")
 
-Compile / doc / scalacOptions ++= Seq("-siteroot", "docs")
-mimaPreviousArtifacts := previousStableVersion.value
-  .map(organization.value %% moduleName.value % _)
-  .toSet
+lazy val examples = project
+  .in(file("slinc-examples"))
+  .dependsOn(slinc)
+  .settings(
+    publishTo := None,
+    publishLocal := Def.task(()),
+    copyNative := {
+      val nativeSrc = (`examples-native` / Compile / nativeCompile).value
+      // rename shared lib name so that it complies with SlinC convention(`{lib name}_{arch}.{ext}`).
+      val destName = {
+        val arch =
+          (`examples-native` / nativePlatform).value.takeWhile(_ != '-')
```

Comment:
nativePlatform returns string in format `{arch}-{kernel}` such as `arm64-darwin`.